pkg/kubelet/network/plugins.go

```diff
@@ -32,12 +32,21 @@ import (
 
 const DefaultPluginName = "kubernetes.io/no-op"
 
+// Called when the node's Pod CIDR is known when using the
+// controller manager's --allocate-node-cidrs=true option
+const NET_PLUGIN_EVENT_POD_CIDR_CHANGE = "pod-cidr-change"
+const NET_PLUGIN_EVENT_POD_CIDR_CHANGE_DETAIL_CIDR = "pod-cidr"
+
 // Plugin is an interface to network plugins for the kubelet
 type NetworkPlugin interface {
 	// Init initializes the plugin.  This will be called exactly once
 	// before any other methods are called.
 	Init(host Host) error
 
+	// Called on various events like:
+	// NET_PLUGIN_EVENT_POD_CIDR_CHANGE
+	Event(name string, details map[string]interface{})
```

Comment:
Do you actually have plans to extend this further?  I like this better, but it feels like YAGNI and the more specific form of it was more descriptive.  I don't feel strongly about it.  This is fine, I guess :)